fuzzers/baby/baby_fuzzer_custom_input/src/input.rs

```diff
@@ -0,0 +1,147 @@
+    pub fn byte_array_mut(&mut self) -> &mut Vec<u8> {
+        &mut self.byte_array
+    }
+
+    /// Returns an immutable reference to the byte array wrapped in [`Some`]
+    pub fn byte_array_optional(&self) -> Option<&Vec<u8>> {
+        Some(&self.byte_array)
+    }
+
+    /// Returns a mutable reference to the optional byte array
+    pub fn optional_byte_array_mut(&mut self) -> &mut Option<Vec<u8>> {
+        &mut self.optional_byte_array
+    }
+
+    /// Returns an immutable reference to the optional byte array
+    pub fn optional_byte_array_optional(&self) -> Option<&Vec<u8>> {
+        self.optional_byte_array.as_ref()
+    }
+}
+
+/// A generator for [`CustomInput`] used in this example
+pub struct CustomInputGenerator {
+    pub max_len: usize,
+}
+
+impl CustomInputGenerator {
+    /// Creates a new [`CustomInputGenerator`]
+    pub fn new(max_len: usize) -> Self {
+        Self { max_len }
+    }
+}
+
+impl<S> Generator<CustomInput, S> for CustomInputGenerator
+where
+    S: HasRand,
+{
+    fn generate(&mut self, state: &mut S) -> Result<CustomInput, Error> {
+        let byte_array = generate_bytes(self.max_len, state);
+        let optional_byte_array = state
+            .rand_mut()
+            .coinflip(0.5)
+            .then(|| generate_bytes(self.max_len, state));
+        let boolean = state.rand_mut().coinflip(0.5);
+
+        Ok(CustomInput {
+            byte_array,
+            optional_byte_array,
+            boolean,
+        })
+    }
+}
+
+/// Generate a [`Vec<u8>`] of a length between 1 (incl.) and `length` (incl.) filled with random bytes
+fn generate_bytes<S: HasRand>(length: usize, state: &mut S) -> Vec<u8> {
+    let rand = state.rand_mut();
+    let len = rand.between(1, length);
+    let mut vec = Vec::new();
+    vec.resize_with(len, || rand.next() as u8);
+    vec
+}
+
+/// [`Mutator`] that toggles the optional byte array of a [`CustomInput`], i.e. sets it to [`None`] if it is not, and to a random byte array if it is [`None`]
+pub struct ToggleOptionalByteArrayMutator {
+    length: usize,
+}
+
+impl ToggleOptionalByteArrayMutator {
+    /// Creates a new [`ToggleOptionalByteArrayMutator`]
+    pub fn new(length: usize) -> Self {
+        Self { length }
+    }
+}
+
+impl<S> Mutator<CustomInput, S> for ToggleOptionalByteArrayMutator
+where
+    S: HasRand,
+{
+    fn mutate(&mut self, state: &mut S, input: &mut CustomInput) -> Result<MutationResult, Error> {
+        input.optional_byte_array = match input.optional_byte_array {
+            None => Some(generate_bytes(self.length, state)),
+            Some(_) => None,
+        };
+        Ok(MutationResult::Mutated)
+    }
+}
+
+impl Named for ToggleOptionalByteArrayMutator {
+    fn name(&self) -> &Cow<'static, str> {
+        &Cow::Borrowed("ToggleOptionalByteArrayMutator")
+    }
+}
+
+/// [`Mutator`] that toggles the boolean field in a [`CustomInput`]
+pub struct ToggleBooleanMutator;
+
+impl<S> Mutator<CustomInput, S> for ToggleBooleanMutator {
```

Comment:
Instead of hard-coding the `CustomInput` here, this could also use a mapper, hmmm

Reply:
I've thought about having default/library mutators booleans, but you'd essentially have to provide a getter and a setter method, and it'd be quite a bit slower. So I figured I'd leave it up to the user to write custom mutators for now, because those have access to the custom input struct directly.

The same applies to toggling `Option<Vec<u8>>` parts of the input btw.

Reply:
getter and setter methods should be optimized away by the compiler anyway, right?

Reply:
Probably, haven't tested it. Can we put this into another issue/PR? Seems only tangentially related.

Also: I'd leave at least one custom mutator in the example fuzzer, just to show that it is possible to add custom mutators and merge them with library-provided ones.

Reply:
Yeah this is only an example so it's fine.